### PR TITLE
Add error collection to handle_includes

### LINKED ISSUE #5314: Add error localization to handle_includes in `lark_parser.py`
1. If there is a cyclical import, the INCLUDE line in the "main" .ert config file should be linted, and thus also localized. The error message. 
2. If too many INCLUDE arguments are given, an error is thrown, also if a file imports another file that does this.

### PR DESCRIPTION
**Pull request by yngve-sk**
And some more logic to properly detect and display cycles

**Issue**
Resolves #5314

**Approach**
Added some more detailed include handling to lark parser to properly detect imports. One consideration here is that ert files are linted once at a time, so when finding errors in other files, they must still be located to their origins in the currently active .ert file.

## Pre review checklist

- [ ] Added appropriate release note label
- [ ] PR title captures the intent of the changes, and is fitting for release notes.
- [ ] Commit history is consistent and clean, in line with the [contribution guidelines](https://github.com/equinor/ert/blob/main/CONTRIBUTING.md).
- [ ] Updated documentation
- [ ] Ensured new behaviour is tested

Adding labels helps the maintainers when writing release notes. This is the [list of release note labels](https://github.com/equinor/ert/labels?q=release-notes).

### CODE REVIEW

File: src/ert/parsing/lark_parser.py [hunk 1]
```diff
@@ -250,18 +252,57 @@ def _substitute_args(
     ]
 
 
+class IncludeNode:
+    def __init__(self, parent: "IncludeNode", value: str):
+        self.parent = parent
+        self.value = value
```
Comment (eivindjahren) [inline src/ert/parsing/lark_parser.py:258]: Does it make sense to call this field `config_file` instead of `value`?

Comment (yngve-sk) [reply src/ert/parsing/lark_parser.py:258]: Yup, changing

File: tests/test_config_parsing/test_parser_error_collection.py [hunk 2]
```diff
@@ -358,6 +415,155 @@ def test_queue_option_max_running_non_int():
...
+INCLUDE include4.ert"""
+                ),
+                "include4.ert": FileDetail(
+                    contents="""
+JOBNAME included
+INCLUDE include5.ert"""
+                ),
+                "include5.ert": FileDetail(
+                    contents="""
+JOBNAME included
```
[the remark this answers is not in the document]
Comment (yngve-sk) [reply tests/test_config_parsing/test_parser_error_collection.py:511]: Yep, adding it

File: src/ert/parsing/lark_parser.py [hunk 3]
```diff
@@ -250,18 +252,57 @@ def _substitute_args(
     ]
 
 
+class IncludeNode:
```
Comment (eivindjahren) [inline src/ert/parsing/lark_parser.py:255]: We have a bit of PSD when it comes to naming things with "Node". Its too general and undescriptive. I suggest we call it `IncludedFile` and rename `parent` to `included_from`.

File: tests/test_config_parsing/test_parser_error_collection.py [hunk 4]
```diff
@@ -341,14 +344,79 @@ def test_summary_without_eclbase():
...
+            line=3,
+            column=19,
+            end_column=44,
+            match=r"Keyword:INCLUDE must have exactly one argument",
+        ),
+    )
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_include_with_too_many_args_error_is_located_indirect(tmpdir):
```
Comment (eivindjahren) [inline tests/test_config_parsing/test_parser_error_collection.py:385]: The name of this becomes a bit unreadable as the `that` is getting in your way. Just remove `that` in this case.

File: tests/test_config_parsing/test_parser_error_collection.py [hunk 5]
```diff
@@ -341,14 +344,79 @@ def test_summary_without_eclbase():
     )
 
 
+@pytest.mark.usefixtures("use_tmpdir")
+def test_include_non_existing_file_is_located(tmpdir):
```
Comment (eivindjahren) [inline tests/test_config_parsing/test_parser_error_collection.py:348]: Just write out the test name here I think `test_that_including_non_existing_file_errors_with_location`

File: tests/test_config_parsing/test_parser_error_collection.py [hunk 6]
```diff
@@ -341,14 +344,79 @@ def test_summary_without_eclbase():
...
+            line=4,
+            column=9,
+            end_column=24,
+            match=r"INCLUDE file:.*does_not_exists not found",
+        ),
+    )
+
+
+@pytest.mark.usefixtures("use_tmpdir")
+def test_that_include_with_too_many_args_error_is_located(tmpdir):
```
Comment (eivindjahren) [inline tests/test_config_parsing/test_parser_error_collection.py:367]: I think `errors_with_location` is slightly more descriptive.

**Review by eivindjahren — APPROVED** [review]
:boat:

**Pull request merged**